gateway-service/src/main/java/org/zowe/apiml/gateway/ws/WebSocketProxyServerHandler.java

```diff
@@ -182,7 +187,17 @@ private void openWebSocketConnection(RoutedService service, ServiceInstance serv
         log.debug(String.format("Opening routed WebSocket session from %s to %s with %s by %s", uri.toString(), targetUrl, webSocketClientFactory, this));
 
         WebSocketRoutedSession session = webSocketRoutedSessionFactory.session(webSocketSession, targetUrl, webSocketClientFactory);
-        routedSessions.put(webSocketSession.getId(), session);
+        session.getWebSocketClientSession().addCallback(
+            successSession -> routedSessions.put(webSocketSession.getId(), session),
+            failure -> {
+                log.debug("Failed opening client web socket session against {}. Server WebSocket session is {}", targetUrl, webSocketSession.getId(), failure);
+                try {
+                    session.getClientHandler().handleMessage(webSocketSession, new TextMessage("Failed opening a session against " + targetUrl + ": " + failure.getMessage()));
+                    webSocketSession.close(CloseStatus.SERVER_ERROR);
+                } catch (Exception e) {
+                    log.debug("Failed seding / closing WebSocket session", e);
```

Comment:
Maybe a `finally` can be added here to make sure the session is closed. 
```suggestion
                    log.debug("Failed seding / closing WebSocket session", e);
                     } finally {
                // Use the existing close method to ensure the session is closed and cleaned up
                close(webSocketSession, CloseStatus.SERVER_ERROR);
            }
```